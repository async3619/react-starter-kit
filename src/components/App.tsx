/**
 * React Starter Kit (https://www.reactstarterkit.com/)
 *
 * Copyright Â© 2014-present Kriasoft, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import React, { ReactNode } from "react";
import StyleContext from "isomorphic-style-loader/StyleContext";

import AppContext, { AppContextTypes } from "../context";

interface Props {
    insertCss: Function;
    context: AppContextTypes;
    children: ReactNode;
}

const App = ({ insertCss, context, children }: Props) => (
    // NOTE: If you need to add or modify header, footer etc. of the app,
    // please do that inside the Layout component.
    <AppContext.Provider value={context}>
        <StyleContext.Provider value={{ insertCss }}>{children}</StyleContext.Provider>
    </AppContext.Provider>
);

export default App;
