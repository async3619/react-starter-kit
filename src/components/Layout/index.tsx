import React from "react";
import withStyles from "isomorphic-style-loader/withStyles";

import Header from "@components/Header";
import Navigation from "@components/Navigation";

import s from "./index.scss";

interface Props {
    children: React.ReactNode;
}

class Layout extends React.Component<Props> {
    public render() {
        const { children } = this.props;

        return (
            <div className={s.root}>
                <Navigation />
                <Header />
                {children}
            </div>
        );
    }
}
export default withStyles(s)(Layout);
