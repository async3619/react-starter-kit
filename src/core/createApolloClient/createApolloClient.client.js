// @flow

import { ApolloClient } from 'apollo-client';
import { from } from 'apollo-link';
import { onError } from 'apollo-link-error';
import { BatchHttpLink } from 'apollo-link-batch-http';
import apolloLogger from 'apollo-link-logger';
import { withClientState } from 'apollo-link-state';
import createCache from './createCache';
import { resolvers as clientSideResolvers } from '../../state';

export default function createApolloClient() {
  const cache = createCache();

  const stateLink = withClientState({
    cache,
    defaults: window.App.initialState,
    resolvers: clientSideResolvers,
  });

  const link = from([
    // batchLink,
    stateLink,
    onError(({ graphQLErrors, networkError }) => {
      if (graphQLErrors)
        graphQLErrors.map(({ message, locations, path }) =>
          console.warn(
            `[GraphQL error]: Message: ${message}, Location: ${locations}, Path: ${path}`,
          ),
        );
      if (networkError) console.warn(`[Network error]: ${networkError}`);
    }),
    ...(__DEV__ ? [apolloLogger] : []),
    new BatchHttpLink({
      uri: '/graphql',
      credentials: 'include',
      batchInterval: 1000,
    }),
  ]);

  return new ApolloClient({
    link,
    cache: cache.restore(window.App.cache),
    queryDeduplication: true,
    connectToDevTools: true,
  });
}
