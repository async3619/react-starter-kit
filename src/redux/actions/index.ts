/* eslint-disable import/prefer-default-export */
import { ActionType } from "typesafe-actions";

import * as RuntimeActionsImpl from "./runtime";

export const RuntimeActions = RuntimeActionsImpl;

export type RuntimeAction = ActionType<typeof RuntimeActions>;
export type RootAction = RuntimeAction;
