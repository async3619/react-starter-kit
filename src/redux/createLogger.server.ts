import { inspect } from "util";
import { Middleware } from "redux";

function inspectObject(object: any) {
    return inspect(object, {
        colors: true,
    });
}

// Server side redux action logger
export default function createLogger(): Middleware {
    // eslint-disable-next-line no-unused-vars
    return () => next => action => {
        let formattedPayload = "";
        if (action.toString !== Object.prototype.toString) {
            formattedPayload = action.toString();
        } else if (typeof action.payload !== "undefined") {
            formattedPayload = inspectObject(action.payload);
        } else {
            formattedPayload = inspectObject(action);
        }

        console.log(` * ${action.type}: ${formattedPayload}`); // eslint-disable-line no-console
        return next(action);
    };
}
