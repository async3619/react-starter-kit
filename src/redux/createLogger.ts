import { Middleware } from "redux";

const createLogger: () => Middleware = (__WEB__ ? require("./createLogger.client") : require("./createLogger.server"))
    .default;

export default createLogger;
