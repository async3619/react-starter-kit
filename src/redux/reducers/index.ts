/* eslint-disable import/prefer-default-export */
import { combineReducers } from "redux";
import { StateType } from "typesafe-actions";

import runtime from "@root/redux/reducers/runtime";

export const rootReducer = combineReducers({
    runtime,
});

export type RootState = StateType<typeof rootReducer>;
