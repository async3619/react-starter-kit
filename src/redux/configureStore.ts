import { applyMiddleware, createStore, DeepPartial, Middleware, StoreEnhancer } from "redux";
import { composeWithDevTools } from "redux-devtools-extension/developmentOnly";

import { rootReducer, RootState } from "@root/redux/reducers";
import createLogger from "@root/redux/createLogger";
import { RootAction } from "@root/redux/actions";

import pkg from "../../package.json";

export default function configureStore(initialState: DeepPartial<RootState>) {
    const middleware: Middleware[] = [];
    let enhancer: StoreEnhancer;

    if (__DEV__) {
        middleware.push(createLogger());

        // https://github.com/zalmoxisus/redux-devtools-extension#14-using-in-production
        const composeEnhancers = composeWithDevTools({
            // Options: https://github.com/zalmoxisus/redux-devtools-extension/blob/master/docs/API/Arguments.md#options
            name: `${pkg.name}@${pkg.version}`,
        });

        // https://redux.js.org/docs/api/applyMiddleware.html
        enhancer = composeEnhancers(applyMiddleware(...middleware));
    } else {
        enhancer = applyMiddleware(...middleware);
    }

    // https://redux.js.org/docs/api/createStore.html
    const store = createStore<RootState, RootAction, {}, {}>(rootReducer, initialState, enhancer);

    // Hot reload reducers (requires Webpack or Browserify HMR to be enabled)
    if (__DEV__ && module.hot) {
        module.hot.accept("./reducers", () =>
            // eslint-disable-next-line global-require
            store.replaceReducer(require("./reducers").default),
        );
    }

    return store;
}

export type StoreType = ReturnType<typeof configureStore>;
