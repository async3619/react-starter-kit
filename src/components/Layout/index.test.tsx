/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from "react";
import renderer from "react-test-renderer";
import { Middleware } from "redux";
import configureStore from "redux-mock-store";

import { RootState } from "@root/redux/reducers";

import App from "../App";
import Layout from ".";
import createApolloClient from "../../utils/createApolloClient.server";

const middlewares: Middleware[] = [];
const mockStore = configureStore<RootState>(middlewares);

describe("Layout", () => {
    test("renders children correctly", () => {
        const client = createApolloClient({} as any, {} as any);
        const store = mockStore({
            runtime: {
                initialTime: 0,
            },
        });

        const wrapper = renderer
            .create(
                <App
                    context={{
                        pathname: "",
                        query: {},
                        client,
                        store: store as any,
                    }}
                    client={client}
                    store={store as any}
                    insertCss={() => {}}
                >
                    <Layout>
                        <div className="child" />
                    </Layout>
                </App>,
            )
            .toJSON();

        expect(wrapper).toMatchSnapshot();
    });
});
