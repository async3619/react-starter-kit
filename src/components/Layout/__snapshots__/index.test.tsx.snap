// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders children correctly 1`] = `
<div
  className="root"
>
  <div
    className="root"
  >
    <div
      className="container"
    >
      <logo.svg
        className="logo"
      />
      <h2
        className="title"
      >
        Create React App
      </h2>
    </div>
  </div>
  <header
    className="root"
  >
    <div
      className="container"
    >
      <logo.svg
        className="logo"
      />
      <h1
        className="title"
      >
        Create React App
      </h1>
      <h2
        className="subtitle"
      >
        Set up a modern web app by running one command.
      </h2>
    </div>
  </header>
  <div
    className="child"
  />
</div>
`;
